fastify.d.ts

```diff
@@ -490,12 +511,12 @@ declare namespace fastify {
      * Hook that is fired after a request is processed, but before the "onResponse"
      * hook
      */
-    addHook(name: 'onSend', hook: (this: FastifyInstance<HttpServer, HttpRequest, HttpResponse>, req: FastifyRequest<HttpRequest>, reply: FastifyReply<HttpResponse>, payload: any, done: (err?: Error, value?: any) => void) => void): FastifyInstance<HttpServer, HttpRequest, HttpResponse>
+    addHook(name: 'onSend', hook: (this: FastifyInstance<HttpServer, HttpRequest, HttpResponse>, req: FastifyRequest<HttpRequest>, reply: FastifyReply<HttpResponse>, payload: any, done: (err?: FastifyError, value?: any) => void) => void): FastifyInstance<HttpServer, HttpRequest, HttpResponse>
 
      /**
      * Hook that is called when a response is about to be sent to a client
      */
-    addHook(name: 'onResponse', hook: (this: FastifyInstance<HttpServer, HttpRequest, HttpResponse>, res: http.ServerResponse, next: (err?: Error) => void) => void): FastifyInstance<HttpServer, HttpRequest, HttpResponse>
+    addHook(name: 'onResponse', hook: (this: FastifyInstance<HttpServer, HttpRequest, HttpResponse>, res: http.ServerResponse, next: (err?: FastifyError) => void) => void): FastifyInstance<HttpServer, HttpRequest, HttpResponse>
```

Comment:
I don’t think this should be a FastifyError

Reply:
If it will be standard Error, then it couldn't be extended (actually it could, but it will affect all Errors, that's not good).

Reply:
I am a bit lost here. You can pass any kind of error with any randomly attached property here. If a user has MyError that descend from Error, it should be able to pass it in here.
That error will end up in the error handler.

Over there maybe you should cast it to whatever it’s needed.

 @delvedor wdyt?

Reply:
Oh, I've just missed that this is a callback. Then it's all fine, there could be Error.